cmd/api-resource-collector/scap.go

```diff
@@ -0,0 +1,235 @@
+// Our initial ugly search trail is:
+//  - find "yamlfilecontent_test" nodes. This is all of them (not filtered by profile)
+//  - for each find the test id and jump to the definition
+//  - find "objectpath" and "file" from the check's ref_id
+// This search trail works with
+//
+//        scanType: Platform
+//        profile: xccdf_org.ssgproject.content_profile_platform-moderate
+//        content: ssg-ocp4-ds.xml
+//        contentImage: quay.io/mrogers950/ocp4-openscap-content:platform
+//
+// TODO: The real check takes selected profile into account and goes something like this (rough notes):
+//  - Find "Profile" with id == selected ID.
+//		- if none, error.
+//	- Find each "selected" child.
+//  - If selected=="true" and idref==prefix(xccdf_org.ssgproject.content_rule_)
+//     - strip off xccdf_org.ssgproject.content_rule_ prefix and save that id.
+// 	- use saved ID to find "reference" with ref_id==id
+func getResourcePaths(ds *utils.XMLDocument) map[string]string {
+	out := map[string]string{}
+
+	LOG("Searching for API paths and filenames from datastream..")
+	nodes := ds.Root.Query("//yamlfilecontent_test")
+	for _, node := range nodes {
+		id := node.GetAttributeValue("id")
+		DBG("yamlfilecontent_test id: '%s'", id)
+		if !strings.HasPrefix(id, "oval:ssg-test_") || !strings.HasSuffix(id, ":tst:1") {
+			continue
+		}
+		idVal := id[len("oval:ssg-test_") : len(id)-len(":tst:1")]
+		DBG("trimmed id value: '%s'", idVal)
+		defDom := ds.Root.Query("//definition")
+		for _, dr := range defDom {
+			metaChild := dr.GetChild("metadata")
+			if metaChild == nil {
+				continue
+			}
+			refChild := metaChild.GetChild("reference")
+			if refChild == nil {
+				continue
+			}
+			refId := refChild.GetAttributeValue("ref_id")
+			if idVal != refId {
+				continue
+			}
+			DBG("reading tags from ref_id '%s'\n", refId)
+			obPath := refChild.GetAttributeValue("objectpath")
+			if obPath == "" {
+				DBG("no 'objectpath' tag")
+				continue
+			}
+			fileName := refChild.GetAttributeValue("file")
+			if fileName == "" {
+				DBG("no 'file' tag")
+				continue
+			}
+			out[obPath] = fileName
+		}
+	}
+	return out
+}
+
+func (c *scapContentDataStream) FetchResources() error {
+	found, err := fetch(c.client, c.resources)
+	if err != nil {
+		return err
+	}
+	c.found = found
+	return nil
+}
+
+func fetch(client *kubernetes.Clientset, objects map[string]string) (map[string][]byte, error) {
+	results := map[string][]byte{}
+	for uri, file := range objects {
+		err := func() error {
+			LOG("Fetching URI: '%s' File: '%s'", uri, file)
+			req := client.RESTClient().Get().RequestURI(uri)
+			stream, err := req.Stream()
+			if err != nil {
+				return err
+			}
+			defer stream.Close()
+			body, err := ioutil.ReadAll(stream)
+			if err != nil {
+				return err
+			}
+			if len(body) == 0 {
+				DBG("no data in request body")
+				return nil
+			}
+			yaml, err := yaml.JSONToYAML(body)
```

Comment:
Instead of this, you could use the `k8s.io/apimachinery/pkg/util/yaml` package, which contains similar functions. Eventually, you can get that into an `unstructured.Unstructured` object which allows us to have a generic way to access the objects.

I used that for the CaC e2e tests if you want a reference: https://github.com/ComplianceAsCode/content/blob/master/tests/ocp4e2e/helpers.go#L392

This way, we also don't introduce another dependency.

Reply:
@JAORMX Will do, thanks!